aci.go

```diff
@@ -758,7 +758,10 @@ func (p *ACIProvider) GetPod(ctx context.Context, namespace, name string) (*v1.P
 	cg, status, err := p.aciClient.GetContainerGroup(ctx, p.resourceGroup, fmt.Sprintf("%s-%s", namespace, name))
 	if err != nil {
 		if status != nil && *status == http.StatusNotFound {
-			return nil, nil
+			// There is bug in vk v1.1 (in Sync loop used as a substitute for async provider updates api) which causes crash if we returned
+			// a (nil, nil) from GetPodStatus() for a newly created pod, will return NotFound instead.
+			// TODO: This loop isn't going to be available beyond v1.1, changes on provider needs to handle it.
```

Comment:
Note we may revert that loop removal for the sake of compatibility.
I think we are able to work around the issues we were having with reconciling graceful pod deletion on sync providers without even requiring special handling, but it still needs to be tested.

Reply:
That sounds great. Anyway, I'm looking actually from ACI side to implement a proper async model for receiving the pods updates. we should anyway try to avoid the 5-sec polling model.
In case the code is added back, can we fix this part (in func (pc *PodController) fetchPodStatusFromProvider()): 
`				podStatus.ContainerStatuses[i].State.Running = nil
			}
		} else if err != nil {
			return nil, nil
		}
	} else if err != nil {
		return nil, err
	}

	pod := podFromKubernetes.DeepCopy()
	podStatus.DeepCopyInto(&pod.Status)` 
to 
`				podStatus.ContainerStatuses[i].State.Running = nil
			}
		} else {                                             //<-- here, so podStatus is not accessed while nil
			return nil, nil
		}
	} else if err != nil {
		return nil, err
	}

	pod := podFromKubernetes.DeepCopy()
	podStatus.DeepCopyInto(&pod.Status)`

I was going to fix it, but found the code being removed :)

BTW, I was planning to publish with v1.0 initially, but the thing that made me switch to v1.1 and hit that, was another issue that happened rarely while i was testing through kubetest. exactly nill access here:

I actually did some testing initially against v1.0, but saw crashes happening in some cases (while running kubetest tests), mainly here:

`panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x125fadd]

goroutine 140 [running]:
github.com/virtual-kubelet/virtual-kubelet/node.(*PodController).updatePodStatus(0xc00007f5c0, 0x18f6520, 0xc0004250e0, 0xc0001c4380, 0x0, 0x0)
	/home/ibabou/go/pkg/mod/github.com/virtual-kubelet/virtual-kubelet@v1.0.0/node/pod.go:251 +0x40d
github.com/virtual-kubelet/virtual-kubelet/node.(*PodController).podStatusHandler(0xc00007f5c0, 0x18f6520, 0xc000424cc0, 0xc001573d80, 0x16, 0x0, 0x0)
	/home/ibabou/go/pkg/mod/github.com/virtual-kubelet/virtual-kubelet@v1.0.0/node/pod.go:307 +0x4b9
github.com/virtual-kubelet/virtual-kubelet/node.handleQueueItem.func1(0xc000507ea0, 0x191ab80, 0xc00050e5c0, 0x18ff7a0, 0xc000652c80, 0xc0000d1ef8, 0x143b820, 0xc00042f440, 0x0, 0x0)
	/home/ibabou/go/pkg/mod/github.com/virtual-kubelet/virtual-kubelet@v1.0.0/node/queue.go:69 +0x22f
github.com/virtual-kubelet/virtual-kubelet/node.handleQueueItem(0x18f6520, 0xc0004249c0, 0x191ab80, 0xc00050e5c0, 0xc0000d1ef8, 0x143b800)
	/home/ibabou/go/pkg/mod/github.com/virtual-kubelet/virtual-kubelet@v1.0.0/node/queue.go:83 +0x1f6
github.com/virtual-kubelet/virtual-kubelet/node.(*PodController).processPodStatusUpdate(0xc00007f5c0, 0x18f6520, 0xc00184f620, 0x16b34ce, 0x1, 0x191ab80, 0xc00050e5c0, 0x0)
	/home/ibabou/go/pkg/mod/github.com/virtual-kubelet/virtual-kubelet@v1.0.0/node/queue.go:116 +0x17f
github.com/virtual-kubelet/virtual-kubelet/node.(*PodController).runProviderSyncWorker(0xc00007f5c0, 0x18f6520, 0xc000565260, 0x16b34ce, 0x1, 0x191ab80, 0xc00050e5c0)
	/home/ibabou/go/pkg/mod/github.com/virtual-kubelet/virtual-kubelet@v1.0.0/node/queue.go:105 +0x6a
github.com/virtual-kubelet/virtual-kubelet/node.(*PodController).runProviderSyncWorkers.func1(0xc00007f5c0, 0x18f6520, 0xc000565260, 0x191ab80, 0xc00050e5c0, 0x3)
	/home/ibabou/go/pkg/mod/github.com/virtual-kubelet/virtual-kubelet@v1.0.0/node/queue.go:99 +0x7f
created by github.com/virtual-kubelet/virtual-kubelet/node.(*PodController).runProviderSyncWorkers
	/home/ibabou/go/pkg/mod/github.com/virtual-kubelet/virtual-kubelet@v1.0.0/node/queue.go:97 +0x79`

Reply:
That's good stuff, can you open issues upstream?

Reply:
I can open an issue, but the code doesn't exist today anymore for sync loop. if you still want me to open an issue, let me know.
For the later issue, this isn't happening with v1.1. we should be good.